samples/net/sockets/echo_server/src/tcp.c

```diff
@@ -118,6 +139,7 @@ static int process_tcp(struct data *data)
 
 		offset += received;
 
+#if !defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
```

Comment:
Well, now we can come back to my comment "why bother to port that legacy code instead of writing normal socket app from scratch?". It related exactly to chunks like this - they don't appear in a normal socket app, and you had to comment them out adding TLS stream support.

Reply:
To be honest I'm not happy with this part of code either and would be very happy to get rid of this - it's more like a workaround code than an actual value to the sample. 

The problem is that current socket shim layer, even if we provide buffer large enough to hold entire message to the `recv` function, will pass it chunk by chunk to the application (or rather net_buf by net_buf). I know it's a stream and this behaviour is correct, still, it's unnecessary overhead. And it doesn't matter whether an app is a port of some legacy code, or a "pure" socket application - the simple `echo` sample behaves the same way.

Maybe I'm a little bit oversensitive here, but I still have Thread somewhere in a back of my head, and hopefully one day will have some time to verify socket apps with OpenThread. And there, such a fragmentation be a huge impact to the throughput.